Mathlib/Analysis/Meromorphic/Basic.lean

```diff
@@ -91,80 +96,98 @@ lemma smul {f : 𝕜 → 𝕜} {g : 𝕜 → E} {x : 𝕜} (hf : MeromorphicAt f
   module
 
 @[fun_prop]
-lemma fun_smul {f : 𝕜 → 𝕜} {g : 𝕜 → E} {x : 𝕜} (hf : MeromorphicAt f x) (hg : MeromorphicAt g x) :
+lemma fun_smul {f : 𝕜 → 𝕜} {g : 𝕜 → E} (hf : MeromorphicAt f x) (hg : MeromorphicAt g x) :
     MeromorphicAt (fun z ↦ f z • g z) x :=
   hf.smul hg
 
 @[deprecated (since := "2025-05-09")] alias smul' := fun_smul
 
 @[fun_prop]
-lemma mul {f g : 𝕜 → 𝕜} {x : 𝕜} (hf : MeromorphicAt f x) (hg : MeromorphicAt g x) :
+lemma mul {f g : 𝕜 → 𝕜} (hf : MeromorphicAt f x) (hg : MeromorphicAt g x) :
     MeromorphicAt (f * g) x :=
   hf.smul hg
 
 @[fun_prop]
-lemma fun_mul {f g : 𝕜 → 𝕜} {x : 𝕜} (hf : MeromorphicAt f x) (hg : MeromorphicAt g x) :
+lemma fun_mul {f g : 𝕜 → 𝕜} (hf : MeromorphicAt f x) (hg : MeromorphicAt g x) :
     MeromorphicAt (fun z ↦ f z * g z) x :=
   hf.smul hg
 
 @[deprecated (since := "2025-05-09")] alias mul' := fun_mul
 
-/-- Finite products of meromorphic functions are analytic. -/
+/-- Finite products of meromorphic functions are meromorphic. -/
 @[fun_prop]
-theorem prod {ι : Type*} {s : Finset ι} {f : ι → 𝕜 → 𝕜} {x : 𝕜}
-    (h : ∀ σ, MeromorphicAt (f σ) x) :
-    MeromorphicAt (∏ n ∈ s, f n) x := by
+theorem prod (h : ∀ σ, MeromorphicAt (F σ) x) :
+    MeromorphicAt (∏ n ∈ s, F n) x := by
   classical
   induction s using Finset.induction with
   | empty =>
-    rw [Finset.prod_empty]
+    simp only [Finset.prod_empty]
     exact analyticAt_const.meromorphicAt
   | insert σ s hσ hind =>
     rw [Finset.prod_insert hσ]
     exact (h σ).mul hind
 
-/-- Finite products of meromorphic functions are analytic. -/
+/-- Finite products of meromorphic functions are meromorphic. -/
 @[fun_prop]
-theorem fun_prod {ι : Type*} {s : Finset ι} {f : ι → 𝕜 → 𝕜} {x : 𝕜}
-    (h : ∀ σ, MeromorphicAt (f σ) x) :
-    MeromorphicAt (fun z ↦ ∏ n ∈ s, f n z) x := by
+theorem fun_prod (h : ∀ σ, MeromorphicAt (F σ) x) :
+    MeromorphicAt (fun z ↦ ∏ n ∈ s, F n z) x := by
   convert prod h (s := s)
   simp
 
+/-- Finite sums of meromorphic functions are meromorphic. -/
 @[fun_prop]
-lemma neg {f : 𝕜 → E} {x : 𝕜} (hf : MeromorphicAt f x) : MeromorphicAt (-f) x := by
+theorem sum (h : ∀ σ, MeromorphicAt (G σ) x) :
+    MeromorphicAt (∑ n ∈ s, G n) x := by
+  classical
+  induction s using Finset.induction with
+  | empty =>
+    simp only [Finset.sum_empty]
+    exact analyticAt_const.meromorphicAt
+  | insert σ s hσ hind =>
+    rw [Finset.sum_insert hσ]
+    exact (h σ).add hind
+
+/-- Finite sums of meromorphic functions are meromorphic. -/
+@[fun_prop]
+theorem fun_sum (h : ∀ σ, MeromorphicAt (G σ) x) :
+    MeromorphicAt (fun z ↦ ∑ n ∈ s, G n z) x := by
+  convert sum h (s := s)
```

Comment:
Does
```suggestion
  simpa using sum h (s := s)
```
work?

Reply:
That does not work for me. I get the error message:

```
Type mismatch: After simplification, term
  sum h
 has type
  MeromorphicAt (∑ n ∈ s, F n) x
but is expected to have type
  MeromorphicAt (fun z ↦ ∑ n ∈ s, F n z) x
```